pandas/io/excel.py

```diff
@@ -68,27 +72,207 @@ def get_writer(engine_name):
+                raise NotImplementedError(msg)
+            reader.load_engine()
+            if _is_url(io):
+                data = _urlopen(io).read()
+                self.book = reader.read_buffer(data)
+            else:
+                self.book = reader.open_workbook(io)
+            self.engine = reader.engine
+            return self.engine, self.book
+
+        # try to determine the reader type based on properties of installed
+        # reader modules.
+        for engine, reader in compat.iteritems(_readers):
+            # only try to import readers that have not been imported before
+            if reader.io_class is None:
+                try:
+                    reader.load_engine()
+                # if the reader is not installed, the user could not have
+                # passed the corresponding reader.io_class, so it is safe
+                # to assume that io is not the current reader
+                except ImportError:
+                    continue
+            # Does the io type match the current available reader?
+            if isinstance(io, reader.io_class):
+                self.engine = engine
+                self.book = io
+                return self.engine, self.book
+            # xlrd has some additional/alternative reading mechanisms:
+            elif engine=='xlrd' and hasattr(io, "read"):
+                xlrd_f = _readers['xlrd']
+                # N.B. xlrd.Book has a read attribute too
+                self.book = xlrd_f.read_buffer(io.read())
+                self.engine = engine
+                return self.engine, self.book
+
+        raise ValueError('Must explicitly set engine if not passing in buffer '
+                         'or path for io.')
+
+
+class XLRDFile(BaseFile):
+    """File reader class for MS Excel spreadsheets (depends on xlrd)
+    """
+    extensions = ['xls', 'xlsx', 'xlsm']
+
+    def __init__(self, engine='xlrd'):
+        # engine, extensions, are defined here, but io_class and open_workbook
+        # are only defined when importing the engine
+        super(XLRDFile, self).__init__(engine=engine)
+
+    def load_engine(self):
+        import xlrd  # throw an ImportError if we need to
+        ver = tuple(map(int, xlrd.__VERSION__.split(".")[:2]))
+        if ver < (0, 9):  # pragma: no cover
+            raise ImportError("pandas requires xlrd >= 0.9.0 for excel "
+                              "support, current version " + xlrd.__VERSION__)
+        else:
+            self._open_workbook = xlrd.open_workbook
+            self.io_class = xlrd.Book
+        return True
+
+    def read_buffer(self, data):
+        """Read from a buffer
+        """
+        return self.open_workbook(file_contents=data)
+
+
+class EZODFFile(BaseFile):
+    """File reader class for ODF spreadsheets (depends on ezodf)
+    """
+    extensions = ['ods']
+
+    def __init__(self, engine='ezodf'):
+        # engine, extensions, are defined here, but io_class and open_workbook
```

Comment:
so just set the engine as as a class attribute, e.g. here would be ``engine='ezodf'``. you don't need the ``__init__`` method in the sub-class then.